website_sale_product_contract_gift/models/sale_order.py

```diff
@@ -91,29 +91,25 @@ def action_confirm(self):
                 )
             )
             existing_partners = self.find_contact_partners(order.partner_shipping_id)
-            if existing_partners:
+            if existing_partners and order.partner_shipping_id != existing_partners:
                 contract_partner = existing_partners[0]
+                order.partner_shipping_id.parent_id = contract_partner
+                # this could be an issue if the existing partner already has
+                # an ongoing subscription, and that the address filled by
+                # the gift giver is different from it. However that would be
+                # a very rare case (offering a subscription to someone already
+                # subscribed) and filling the wrong adress.
+                # Besides, the old adress is kept, so the error is easy to fix.
+
```

Comment:
```suggestion
            if existing_partners and order.partner_shipping_id != existing_partners:
                # We move the shipping address provided by the donor
                # to the receiver. The address on the receiver and the
                # one given by the donor can differ. In such case, and
                # when the receiver has an ongoing subscription, the
                # address of the receiver for all its contract will
                # change to the address given by the donor. This can
                # be wrong. But offering a subscription to someone
                # already subscribed AND filling the wrong address is
                # quite rare. Besides, the old address is kept on the
                # receiver. So the error is easy to fix.
                contract_partner = existing_partners[0]
                order.partner_shipping_id.parent_id = contract_partner


```
Proposing a rephrase of the comment.